### PR TITLE
Design Review Update - Complete Your User Profile (No Associations)

### LINKED ISSUE #156: [Task] Style the disabled email input as text
This will match the figma because the email field is no longer a disabled input but a text with the user's email

### LINKED ISSUE #308: [Task] [Complete Your User Profile] Add the 'Provide any additional details' component; add to the submission

### LINKED ISSUE #310: [Design Review] Complete Your User Profile (No Associations)
- [x] Update to Figma text - Text Intro and Section Intro
- [ ] (pending) 8 or 10px underneath email address
- [x] period before quotation marks
- [x] Above the Add button, 30 px
- [x] Above Footer, 60px margin top
- [x] remove the bottom margin of the button group (submit, clear form)
- [x] Below Section Intro, should be 30px
- [x] FormButtonGroup space between buttons should be 10px (remove inline block property of the buttons)
^ Create a DSR issue for the DS and then DSR - Destructive Action

### PR DESCRIPTION
**Pull request by shindigira**
closes #156 
closes #308
closes #310 

## Additional Changes (Updated 04/01/2024)
- content(Complete User Profile): period before quote
- style(CPF No Assoc - FormErrorHeader): FormErrorHeader above the `Step1FormHeader`
- content(Step1FormHeader): update subheading content for Complete user profile with email associations and without email associations
- **fix(CPF No Assoc): added additional_details to post request**
- Removed unused files: `Uploader`, `FilingApp` and `Home`
- **style(additionalDetails - textArea): added .a-text-input class (padding) and reduced rows to 4**

## Additional Changes (Updated 03/28/2024)
- chore(LEI): The Zod check for LEI is now changed to -- A-Z and 0-9 (20 characters)
- feat(Complete User Profile): Added a 'Provide Any Additional Details' section
- feat(FormErrorHeader): Added a `keyLogicFunc` prop that uses logic to choose keys (**used for the scroll links**)
- style(FormHeaderWrapper): Removed marginTop prop since it is handled in `FormWrapper`
- **style(FormWrapper): Created a `marginTop` prop that is used if a **CrumbTrail is used** -- to set a smaller top margin**
- style(FormWrapper): Re-added bottom margin of 60px
- style(InputEntry - Email): Utilizes the `DisplayField` component to **display user's email address**.

## Changes
- style(Fieldgroup): Adhere's to the 770px max-width with box-border (includes borders, margins and padding into the max-width)
- style(SectionIntro): Adhere's to the 670px max-width with box-border (includes borders, margins and padding into the max-width), 30px margin-bottom
- ~~style(FooterWrapper): Footer has 60px margin-top, removed margin-bottom from FormWrapper~~
- style(FormButtonGroup): 10px between buttons
- style(AddFinancialButton): 32.3px ---> 30px margin-bottom
- content(Complete User Profile): Updated content to match the Figma, **period before the ending quotation mark**.

## How to Test (Updated 03/28/2024)
- Test **all** pages (Update the Financial Profile, Point of Contact, Create Profile Form, Create Profile Form - No Associations, Summary, Types of Financial Institution) where `FormWrapper` is used and verify if the **proper top margin** is correct
- FormErrorHeader: Ensure clicking on links and scrolling to the field with the error are functioning properly for both the **Create Profile Form - No Associations** and **Update the Financial Profile** forms.

## Screenshot (Updated 03/28/2024)
Screenshot 2024-03-28 at 7 33 08 PM

### CODE REVIEW

**Review by meissadia — DISMISSED** [review]
Styles look good 👍🏾 
 
I made one small change to update the following content to match the design file where it reads "Once we have associated your user profile".

Before screenshot:
Screenshot 2024-03-18 at 10 12 03 AM

**Review by billhimmelsbach — COMMENTED** [review]
Hey, don't want to barge into this PR, but I'm just trying to get everything moved into main. This looks good to me, except are you planning @shindigira on including the "Provide any additional details" section, importing it from Meissa's work? It'll be hard for Natalia to do a final approval without it.

**Review by billhimmelsbach — CHANGES_REQUESTED** [review]
I went over the email-to-case forms with @natalia-fitzgerald today, and in addition to adding the "Provide any additional details" section I [mentioned](https://github.com/cfpb/sbl-frontend/pull/311#pullrequestreview-1958056975) a few days ago, here are a few other minor design changes we discussed @shindigira:

Screenshot 2024-03-26 at 9 11 12 PM
- too much space above the breadcrumb (take a peek at the figma)

Screenshot 2024-03-26 at 9 11 08 PM
- `There was a problem completing your profile` should be `There was a problem completing your user profile`
- Clicking on `Enter your financial institution's name (1)`, `Enter your financial institution's Legal Entity Identifier (LEI) (1)`, etc doesn't scroll the error field into view

- LEIs are only going to be all caps, so you can drop the `a-z` part in `Must be 20 characters and only contain a-z, A-Z, and 0-9 (no special characters)`

Looking good so far, and excited to finish off these forms. 👍

**Comment by billhimmelsbach** [comment]
Hi @natalia-fitzgerald! So I told @shindigira to fix everything except the clicking one of these errors and scrolling to the form field, since a solution needs to be worked out between Sigmund and Meissa. 

Screenshot 2024-03-28 at 3 44 45 PM



@shindigira will ping us when the rest of the design is fixed, but we'll save fixing the scroll logic for a later PR. 👍

File: src/components/FooterWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { FooterCfGov } from 'design-system-react';
+
+function FooterWrapper(): JSX.Element {
+  return (
+    <div className='mt-[3.75rem]'>
+      <FooterCfGov />
```
Comment (billhimmelsbach) [inline src/components/FooterWrapper.tsx:6]: Heyo, this causes some weirdness @shindigira by pushing up all the content, like here on the unauthenticated landing page:

_Before_
Screenshot 2024-03-28 at 5 14 04 PM


_After_
Screenshot 2024-03-28 at 5 14 15 PM

It also increases the spacing at the bottom too much for other pages:
Screenshot 2024-03-28 at 5 19 29 PM

**Review by billhimmelsbach — CHANGES_REQUESTED** [review]

**Comment by billhimmelsbach** [comment]
The preview site has been updated with the latest from this PR @natalia-fitzgerald, if you wouldn't mind checking the Complete your user profile (no email associations) again?

**Review by natalia-fitzgerald — APPROVED** [review]
@shindigira 
This looks great! Nice work. 

Can you open a separate PR for these additional changes: 
- style the `<html>` so that the color of the background outside the content is gray for all pages (see screenshot at the bottom of the page)
- Text content in the form-level alert box should be 670px wide (see screenshot - text is currently extending the full width of the box)

Screenshot 2024-04-02 at 12 29 43 PM

**Review by billhimmelsbach — APPROVED** [review]

**Pull request merged**